File: springboot-mustache-crud/src/main/java/com/knf/dev/demo/springbootjpamustachecrud/service/UserService.java
package com.knf.dev.demo.springbootjpamustachecrud.service;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import com.knf.dev.demo.springbootjpamustachecrud.model.User;
import com.knf.dev.demo.springbootjpamustachecrud.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import javax.persistence.EntityNotFoundException;

@Service
public class UserService {
    @Autowired
    UserRepository repository;

    public List<User> getAllusers() {
        List<User> result = (List<User>) repository.findAll();
        if (result.size() > 0) {
            return result;
        } else {
            return new ArrayList<User>();
        }
    }

    public User getUserById(Long id)
              throws EntityNotFoundException {

        Optional<User> user = repository.findById(id);
        if (user.isPresent()) {
            return user.get();
        } else {
            throw new EntityNotFoundException
                    ("No user record exist for given id");
        }
    }

    public User createOrUpdateUser(User entity) {
        if (entity.getId() == null) {
            entity = repository.save(entity);
            return entity;
        } else {
            Optional<User> user = repository
                       .findById(entity.getId());
            if (user.isPresent()) {
                User newEntity = user.get();
                newEntity.setEmail(entity.getEmail());
                newEntity.setFirst_name(entity.getFirst_name());
                newEntity.setLast_name(entity.getLast_name());
                newEntity = repository.save(newEntity);
                return newEntity;
            } else {
                entity = repository.save(entity);
                return entity;
            }
        }
    }

    public void deleteUserById(Long id)
             throws EntityNotFoundException {

        Optional<User> user = repository.findById(id);
        if (user.isPresent()) {
            repository.deleteById(id);
        } else {
            throw new EntityNotFoundException
                     ("No user record exist for given id");
        }
    }
}

File: kotlin-spring-main/kotlinspringjpafreemarkercrud/src/main/kotlin/com/knf/dev/demo/kotlinspringjpafreemarkercrud/service/UserService.kt
package com.knf.dev.demo.kotlinspringjpafreemarkercrud.service

import com.knf.dev.demo.kotlinspringjpafreemarkercrud.model.User
import org.springframework.beans.factory.annotation.Autowired
import com.knf.dev.demo.kotlinspringjpafreemarkercrud.repository.UserRepository
import org.springframework.stereotype.Service
import java.util.ArrayList
import kotlin.Throws
import javax.persistence.EntityNotFoundException

@Service
class UserService {
    @Autowired
    var repository: UserRepository? = null
    val allusers: List<User?>
        get() {
            val result = repository!!.findAll() as List<User?>
            return if (result.size > 0) {
                result
            } else {
                ArrayList()
            }
        }

    @Throws(EntityNotFoundException::class)
    fun getUserById(id: Long): User {
        val user = repository!!.findById(id)
        return if (user.isPresent) {
            user.get()
        } else {
            throw EntityNotFoundException("No user record exist for given id")
        }
    }

    fun createOrUpdateUser(entity: User): User {
        var entity = entity
        return if (entity.id == null) {
            entity = repository!!.save(entity)
            entity
        } else {
            val user = repository!!.findById(
                entity.id!!
            )
            if (user.isPresent) {
                var newEntity = user.get()
                newEntity.email = entity.email
                newEntity.firstName = entity.firstName
                newEntity.lastName = entity.lastName
                newEntity = repository!!.save(newEntity)
                newEntity
            } else {
                entity = repository!!.save(entity)
                entity
            }
        }
    }

    @Throws(EntityNotFoundException::class)
    fun deleteUserById(id: Long) {
        val user = repository!!.findById(id)
        if (user.isPresent) {
            repository!!.deleteById(id)
        } else {
            throw EntityNotFoundException("No user record exist for given id")
        }
    }
}